cmake/public/cuda.cmake

```diff
@@ -65,69 +65,13 @@ if(NOT CMAKE_CUDA_COMPILER_VERSION STREQUAL CUDAToolkit_VERSION OR
                       "V${CUDAToolkit_VERSION} in '${CUDAToolkit_INCLUDE_DIR}'")
 endif()
 
-if(NOT TARGET CUDA::nvToolsExt)
-  message(FATAL_ERROR "Failed to find nvToolsExt")
```

Comment:
Why remove this error message?

Reply:
I removed them when I tried to solve the nvtx link problem, they should be recovered.

Reply:
If this check is failing then how is the build succeeding? That doesn't sound right at all.

Reply:
So I remove the nvtoolsext check and link nvToolsExt privately. So that it passes when including cuda.cmake and doesn't actually use nvToolsExt.

Reply:
In other words, nvtoolsext is set to NOTFOUND in cpu case but isn't used at all.

Reply:
This is happening because you're picking the wrong `FindCUDAToolkit.cmake`, see below comment.

Reply:
The version of fundcudatoolkit doesn't matter. In a cpu only test case, the cuda files will be removed after building.

Reply:
`cmake/public/cuda.cmake` assumes too many header dependencies which are mostly not needed to build an extension. So I'm fine with not searching for NVTX headers while building extensions. Also `nvtx` is a header only library now